src/opensearch.py

```diff
@@ -147,30 +147,24 @@ def start(self):
         except OpenSearchCmdError:
             raise OpenSearchStartError()
 
-        retries = 0
-        while not self.is_started() and retries < 3:
-            time.sleep(2)
-            retries += 1
+        start = datetime.now()
+        while not self.is_started() and (datetime.now() - start).seconds < 60:
+            time.sleep(3)
         else:
-            raise OpenSearchStartError()
+            raise OpenSearchStartTimeoutError()
 
     @override
     def _stop_service(self):
         """Stop opensearch."""
-        self._run_cmd(
-            "setpriv",
-            "--clear-groups --reuid ubuntu --regid ubuntu -- sudo systemctl stop opensearch.service",
-        )
-
-        retries = 0
-        while retries < 3:
-            if not self.is_started():
-                return
-
-            time.sleep(2)
-            retries += 1
+        try:
+            self._run_cmd("sudo systemctl stop opensearch.service")
```

Comment:
I don't think that sudo is needed.
btw, there are library for such actions `lib/charms/operator_libs_linux/v1/systemd.py`
```suggestion
            self._run_cmd("systemctl stop opensearch.service")
```